lib/components/base-components/NormalComponent/footprintUtils.ts

```diff
@@ -0,0 +1,20 @@
+/**
```

Comment:
just put into their own files, i wish i could have a linting rule but these files that put together multiple utils cause tons of problems with searching (you can actually see AI take longer whenever they're used because it does a file search then a grep)

This should be two files: `isFootprintUrl.ts` and `parseLibraryFootprintRef.ts`